hive-metastore/src/test/java/org/apache/iceberg/hive/TestHiveCatalog.java

```diff
@@ -559,4 +556,32 @@ public void testSetSnapshotSummary() throws Exception {
     spyOps.setSnapshotSummary(parameters, snapshot);
     Assert.assertEquals("The snapshot summary must not be in parameters due to the size limit", 0, parameters.size());
   }
+
+  @Test
+  public void testSetDefaultPartitionSpec() throws Exception {
+    Schema schema = new Schema(
+        required(1, "id", Types.IntegerType.get(), "unique ID"),
+        required(2, "data", Types.StringType.get())
+    );
+    PartitionSpec spec = PartitionSpec.builderFor(schema)
+        .bucket("data", 16)
+        .build();
+    TableIdentifier tableIdent = TableIdentifier.of(DB_NAME, "tbl");
+
+    try {
+      catalog.buildTable(tableIdent, schema)
+          .withPartitionSpec(spec)
+          .create();
+
+      Assert.assertEquals(PartitionSpecParser.toJson(spec),
```

Comment:
[minor] how about asserting the case where we don't set default partition spec un-partitioned spec